### ISSUE TITLE
QML - Cardiology Application 

### ISSUE THREAD

**Issue by dsierrasosa** [qiskit-advocate/qamp-fall-22#20]
### Description

Cardiology has been one of the branches of medicine where the most progress has been made in recent years. The high prevalence of cardiovascular pathologies has led to a constant development of new technologies for diagnosis and treatment. The most prevalent cardiovascular pathology is that related to the so-called coronary disease, the one that can end up causing the appearance of an acute myocardial infarction. Not surprisingly, it is in this field that some of the most important progress has been made.

This project aims to explore the use of quantum computing to implement Quantum Machine Learning Techniques on ischemic heart disease datasets, taking into account the limitations imposed by the current NISQ devices such as the reduced number of qubits available and the coherence time. In particular, this project is intended to study supervised and unsupervised machine learning algorithms. 

### Deliverables

Tutorial summarizing the findings 

### Mentors details

- Mentor 
  - Name: Daniel Sierra-Sosa
  - GitHub ID: @dsierrasosa
  - What they do: Assistant Professor

### Number of mentees

1

### Type of mentees

- Mentee 1
  - Required:
    -  Time to work in the project
  - Nice to have:
    - Experience in Machine Learning
    - Attended 2021 Qiskit Global Summer School on Quantum Machine Learning

**Comment by sergiomtzlosa** [comment]
Hi @dsierrasosa,

I am Sergio. This seems interesting to me. Maybe, we can work together on this issue. 

Thanks!

**Comment by bopardikarsoham** [comment]
Hello, @dsierrasosa
Soham here. I'm really excited about this project. I would like to chat more regarding this project?

**Comment by pdc-quantum** [comment]
Hi, @dsierrasosa,
I am Pierre, Qiskit advocate and cardiologist. Very excited about this project, as you may figure out. And ready to work in a team.

**Comment by dsierrasosa** [comment]
Hello Everyone! Thank you for your interest in this particular problem, we have conducted an exploratory work on this topic and it looks challenging! If you have time we could meet briefly to talk about what we have done and your expectations with this QAMP.

**Comment by pdc-quantum** [comment]
Sure, I have time to devote to this topic. my local time is CEST (edited). What do you suggest for a meeting?

**Comment by sergiomtzlosa** [comment]
I would be glad to join the meeting to find out more details and decide if the issue fits to me, my timezone is CEST. Thanks a lot!

**Comment by bopardikarsoham** [comment]
Sue would love to have a meet. My timezone is IST(UTC+5:30)

**Comment by dsierrasosa** [comment]
Wow that is a diversity of time zones this is my proposal:
Friday at:
9:30 EST
14:30 UTC +1
15:30 CEST
19:00 UTC +5:30

Is it ok? Do you want me to send the invitation?

**Comment by Alfaxad** [comment]
Hello @dsierrasosa ,  my name is Alfaxad, Qiskit advocate from Tanzania experienced in Machine Learning and Deep Learning, I am interested in this project. My time zone is UTC+3 therefore I will be able to attend the meeting on Friday according to the allocated time.

**Comment by bopardikarsoham** [comment]
I'm ok with the allocated time @dsierrasosa. Thanks a lot!

**Comment by sergiomtzlosa** [comment]
@dsierrasosa I am fine with the schedule, I'll be there, thanks!

**Comment by pdc-quantum** [comment]
@dsierrasosa  I'll be there on Friday

**Comment by hsanthan** [comment]
@dsierrasosa , My name is Hema. I have done my Masters in Data Science and excited about this project. I would like to join the team if there is still room and you think I fit your requirements. I am in the Eastern timezone and happy to join the Friday call. Please share the meeting invite. Thanks!

**Comment by MiikaVuorio** [comment]
@dsierrasosa I too am very excited about this project and would love to join the friday meeting :)

**Comment by dsierrasosa** [comment]
Daniel Sierra-Sosa is inviting you to a scheduled Zoom meeting.

Topic: QAMP-Fall-22
Time: Aug 19, 2022 09:30 AM Eastern Time (US and Canada)

Join Zoom Meeting
https://zoom.us/j/93908685161?pwd=UmI1bElyeE1yenVyOHdqTXU1anMzQT09

Meeting ID: 939 0868 5161
Passcode: 626654
One tap mobile
+13017158592,,93908685161#,,,,*626654# US (Washington DC)
+16469313860,,93908685161#,,,,*626654# US

Dial by your location
        +1 301 715 8592 US (Washington DC)
        +1 646 931 3860 US
        +1 309 205 3325 US
        +1 312 626 6799 US (Chicago)
        +1 646 876 9923 US (New York)
        +1 719 359 4580 US
        +1 253 215 8782 US (Tacoma)
        +1 346 248 7799 US (Houston)
        +1 386 347 5053 US
        +1 408 638 0968 US (San Jose)
        +1 564 217 2000 US
        +1 669 444 9171 US
        +1 669 900 6833 US (San Jose)
Meeting ID: 939 0868 5161
Passcode: 626654
Find your local number: https://zoom.us/u/af4wGa1iy

**Comment by dsierrasosa** [comment]
Please DM me in slack for the Discord link if you are interested in this project! Thanks for joining today.

**Comment by poig** [comment]
@dsierrasosa hi, I am interested in this project, my discord `Jun Liang Tan#4881`

**Comment by hamzakamel1** [comment]
I'm very interesting about this project, it looks really good

**Comment by poig** [comment]
.

**Comment by bopardikarsoham** [comment]
Checkpoint 1 Presentation:

- [Project 20 QAMP Fall 2022 Checkpoint 1.pptx](https://github.com/qiskit-advocate/qamp-fall-22/files/9737049/Project.20.QAMP.Fall.2022.Checkpoint.1.pptx)

- [Project 20 QAMP Fall 2022 Checkpoint 1.pdf](https://github.com/qiskit-advocate/qamp-fall-22/files/9737050/Project.20.QAMP.Fall.2022.Checkpoint.1.pdf)

**Comment by pdc-quantum** [comment]
![qamp-fall-22-qml-cardiology-application](https://user-images.githubusercontent.com/29145045/206029597-4abe7c43-2379-43cd-ab2f-eda9132a90f5.png)

**Comment by dsierrasosa** [comment]
In this Qiskit Advocate Mentorship Project we have been working on applying Quantum Machine Learning (QML) techniques to Cardiology. We have been exploring different techniques for both images and tabular data, in the imaging portion of the project we are predicting Cardiomegaly on X-ray images, and on the tabular data we are predicting Coronary artery disease using 11 clinical features. For the execution of this project, we are using publicly available datasets. 

**IMAGING APPROACH:** 
**Subject:** Detection of cardiomegaly on posteroanterior chest X-ray views by hybrid classical-quantum neural networks
**Comparison:** Hybrid QNN Models vs Classical CNN models
**Dataset sources:** NIH, ChexPert
**Final dataset:**  Final Balanced dataset, mined from CheXpert, and corrected for erroneous labels
**First Approach:** Classical, pretrained with Imagenet (Densenet121, Resnet50, …)
**Quantum Approach:** Exploration of a variety of ansatz to encode data, as in ongoing proposals of the mentees
**Ongoing experimental classical platforms:** Pytorch, Tensorflow
**Final classical platform:** to be determined
**Ongoing experimental quantum platforms:** Qiskit and Pennylane 
**Final quantum platform:** Qiskit
**Target Objective 1:** Similar or better for metrics than classical (AUC ROC score, accuracy, F1-score, precision, recall)
**Target 2:** Minimize computational resources

**TABULAR APPROACH:** 
**Subject:** Detection of Coronary Artery Disease by exploring different encodings and QML methods 
**Comparison:** Classical Machine Learning Performance with QSVM and VQC under different encodings
**Final Dataset:** IEEE DataPort Heart Disease Dataset (Comprehensive)
**First Approach:** ZZFeatureMap, TwoLocal and Custom Ansatz for evaluate performance.
**Second Approach:**  Exploration of Tensor Networks MPS, MERA, TTN. 
**Current Activities:** Exploration of a variety of ansatz to encode data, as in ongoing proposals of the mentees
**Ongoing experimental classical platforms:** Pytorch, Scikit-Learn
**Ongoing experimental quantum platforms:** Qiskit 
**Target Objective 1:** Similar or better for metrics than classical (AUC ROC score, accuracy, F1-score, precision, recall)
**Target 2:** Evaluate different encoding performance

**Comment by bopardikarsoham** [comment]
Final Checkpoint Presentation:
- [Project 20 QAMP Fall 2022 Final Checkpoint.pptx](https://github.com/qiskit-advocate/qamp-fall-22/files/10225950/Project.20.QAMP.Fall.2022.Final.Checkpoint.pptx)

-  [Project 20 QAMP Fall 2022 Final Checkpoint.pdf](https://github.com/qiskit-advocate/qamp-fall-22/files/10225954/Project.20.QAMP.Fall.2022.Final.Checkpoint.pdf)

**Comment by GemmaDawson** [comment]
Congratulations on completing all the requirements for QAMP Fall 2022!! 🌟🌟🌟
